test/suites/scale/deprovisioning_test.go

```diff
@@ -92,11 +92,41 @@ var _ = Describe("Deprovisioning", Label(debug.NoWatch), Label(debug.NoEvents),
 	})
 
 	Context("Multiple Deprovisioners", func() {})
-	Context("Consolidation", func() {})
+	Context("Consolidation", func() {
+		It("should consolidate nodes to get a higher utilization", func() {
+			replicasPerNode := 20
+			maxPodDensity := replicasPerNode + dsCount
+			expectedNodeCount := 200
+			replicas := replicasPerNode * expectedNodeCount
+
+			deployment.Spec.Replicas = lo.ToPtr[int32](int32(replicas))
+			provisioner.Spec.KubeletConfiguration = &v1alpha5.KubeletConfiguration{
+				MaxPods: lo.ToPtr[int32](int32(maxPodDensity)),
+			}
+
+			By("waiting for the deployment to deploy all of its pods")
+			env.ExpectCreated(deployment)
+			env.EventuallyExpectPendingPodCount(selector, replicas)
+
+			By("kicking off provisioning by applying the provisioner and nodeTemplate")
+			env.ExpectCreated(provisioner, nodeTemplate)
+
+			env.EventuallyExpectCreatedMachineCount("==", expectedNodeCount)
+			env.EventuallyExpectCreatedNodeCount("==", expectedNodeCount)
+			env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+			env.EventuallyExpectHealthyPodCount(selector, replicas)
+
+			replicas = int(float64(replicas) * 0.2)
+			deployment.Spec.Replicas = lo.ToPtr[int32](int32(replicas))
+			env.ExpectUpdated(deployment)
+
+			provisioner.Spec.Consolidation = &v1alpha5.Consolidation{Enabled: lo.ToPtr(true)}
+			env.ExpectUpdated(provisioner)
```

Comment:
Should have some assertion of an eventually so that this test doesn't just end immediately after updating the provisioner.  

Maybe this is good in the meantime? 
```suggestion
			env.ExpectUpdated(provisioner)
			env.EventuallyExpectDeletedNodeCount("==", createdNodes * 0.2)
```

Reply:
Yep, this is still in draft so I hadn't had the chance to add it yet. Added now!